lib/Decompressor.js

```diff
@@ -219,10 +267,20 @@ export class Decompressor extends Transformer {
   }
 
   _transformTypedValue({entries, termType, value}) {
-    const DecoderClass = TYPE_DECODERS.get(termType);
-    const decoder = DecoderClass && DecoderClass.createDecoder(
-      {value, transformer: this});
-    if(decoder) {
+    let decoder;
+    if(this.reverseTypedLiteralTable.has(termType)) {
```

Comment:
Hmm, it seems that we check the reverse literal table here and then pass that table into the decoder which just checks it again (maybe multiple times?). It seems to me that `createDecoder` should just run this check for us once and return `false`/whatever the pattern is -- so we don't have to keep going through map lookups for the same thing.